massa-network-exports/src/commands.rs

```diff
@@ -1,16 +1,170 @@
+//!    |               |                         #            
+//!    |               |<------------------------+             `NetworkCommand::SendOperations`
+//!    |<--------------+                         |             `NodeCommand::SendOperations`
+//!    |               |                         |
+//! ```
 
-use crate::{BootstrapPeers, Peers};
-use massa_models::SignedEndorsement;
-use massa_models::SignedHeader;
-use massa_models::SignedOperation;
-use massa_models::{composite::PubkeySig, node::NodeId, stats::NetworkStats, Block, BlockId};
+use crate::{BootstrapPeers, ConnectionClosureReason, Peers};
+use massa_models::{
+    composite::PubkeySig,
+    node::NodeId,
+    operation::{OperationIds, Operations},
+    stats::NetworkStats,
+    Block, BlockId, SignedEndorsement, SignedHeader,
+};
+use std::{collections::HashMap, net::IpAddr};
 use tokio::sync::oneshot;
 
+#[derive(Clone, Debug)]
+pub enum NodeCommand {
+    /// Send given peer list to node.
+    SendPeerList(Vec<IpAddr>),
+    /// Send that block to node.
+    SendBlock(Block),
+    /// Send the header of a block to a node.
+    SendBlockHeader(SignedHeader),
+    /// Ask for a block from that node.
+    AskForBlocks(Vec<BlockId>),
+    /// Close the node worker.
+    Close(ConnectionClosureReason),
+    /// Block not found
+    BlockNotFound(BlockId),
+    /// Send full Operations (send to a node that previously asked for)
+    SendOperations(Operations),
+    /// Send a batch of operation ids
+    SendOperationBatch(OperationIds),
+    /// Ask for a set of operations
+    AskForOperations(OperationIds),
+    /// Endorsements
+    SendEndorsements(Vec<SignedEndorsement>),
+}
+
+/// Event types that node worker can emit
+/// Append on receive something from inside and outside.
+/// Outside init with `Received` prefix.
+#[derive(Clone, Debug)]
+pub enum NodeEventType {
+    /// Node we are connected to asked for advertised peers
+    AskedPeerList,
+    /// Node we are connected to sent peer list
+    ReceivedPeerList(Vec<IpAddr>),
+    /// Node we are connected to sent block
+    ReceivedBlock(Block),
+    /// Node we are connected to sent block header
+    ReceivedBlockHeader(SignedHeader),
+    /// Node we are connected to asks for a block.
+    ReceivedAskForBlocks(Vec<BlockId>),
+    /// Didn't found given block,
+    BlockNotFound(BlockId),
+    /// Operation
+    ReceivedOperations(Operations),
+    /// Received operation batch
+    ReceivedOperationBatch(OperationIds),
+    /// Receive a list of wanted operations
+    ReceivedAskForOperations(OperationIds),
+    /// Receive a set of endorsement
+    ReceivedEndorsements(Vec<SignedEndorsement>),
+}
+
+/// Events node worker can emit.
+/// Events are a tuple linking a node id to an event type
+#[derive(Clone, Debug)]
+pub struct NodeEvent(pub NodeId, pub NodeEventType);
+
 /// Commands that the worker can execute
 #[derive(Debug)]
 pub enum NetworkCommand {
```

Comment:
Even if I think it's not related to this PR I have added documentations of most of them. I didn't make for all of them because some are not clear for me.